doc/adr/2023-05-03-patch-releases-are-ABI-compatible.md

```diff
@@ -0,0 +1,20 @@
+**Title**: Patch releases **must** be ABI compatible.
+
+**Status**: accepted
+
+**Context**: We create a few patch releases each year. These are always bug
+fixes. We have never introduced API changes in a patch release, nor should we.
+However, until this ADR is adopted our patch releases do change the ABI, because
+the patch number is embedded in the inline namespace
+`google::cloud::v$Major_$Minor_$Patch`.
+
+**Decision**: We will remove the patch number from our inline namespace. If we
+ever have to hotfix a bug in such a way that it breaks the API or ABI we will
+**not** use a patch release for such fixes. We will just use a new minor
+release.
+
+**Consequences**: Applications will not be able to link two patch versions. That
```

Comment:
"two patch versions of the same minor release." ?

The example in the next sentence makes it clear what you mean. But I think the sentence, as written, is false.